chalice/app.py

```diff
@@ -350,6 +387,8 @@ def to_dict(self):
         # We want the output of `to_dict()` to be
         # JSON serializable, so we need to remove the CaseInsensitive dict.
         copied['headers'] = dict(copied['headers'])
+        if copied['query_params'] is not None:
+            copied['query_params'] = dict(copied['query_params'])
```

Comment:
Yes. Right now its not possible. I tried to make the change in the least intrusive way possible. I agree that this isn't perfect.
I thought about two other options that maybe could be better but are breaking:
1. Serialize every value as an array of values. This will break current usages, so maybe add a flag?
2. A less intrusive way but a little inconsistent could be serializing only multiple values as lists.

Reply:
What would the flag look like? I agree none of the options are ideal, and the way you have it now is reasonable to avoid breaking anyone. We may need to just bench this until we have some modeled types.